onmt/dynamic/opts.py

```diff
@@ -0,0 +1,99 @@
+"""All options for dynamic running. Should be given in a yaml file."""
+from onmt.opts import _train_general_opts, config_opts, model_opts
+from onmt.dynamic.transforms import AVAILABLE_TRANSFORMS
+
+
+def data_config_opts(parser):
+    """Options related to dynamic data config file."""
+    parser.add('-data_config', '--data_config', required=True,
+               is_config_file_arg=True, help='data config file path')
+    parser.add('-save_data_config', '--save_data_config', required=False,
+               # is_write_out_config_file_arg=True,
```

Comment:
Just a copy-paste from `config_opts` not ever used this option my own. The commented part is needed if we want dump automatically the parsed config, but this will raise a conflict with already existing `-save_config` option. I'll simply remove this as preprocess & train have already merged.